corehq/motech/rate_limiter.py

```diff
@@ -76,12 +76,13 @@ def _get_per_user_repeater_wait_milliseconds_rate_definition(domain):
 @run_only_when(SHOULD_RATE_LIMIT_REPEATERS)
 @silence_and_report_error("Exception raised in the repeater rate limiter",
                           'commcare.repeaters.rate_limiter_errors')
-def rate_limit_repeater(domain):
-    is_domain_allowed_usage = repeater_rate_limiter.allow_usage(domain)
-    if RATE_LIMIT_REPEATER_ATTEMPTS.enabled(domain, namespace=NAMESPACE_DOMAIN):
-        is_domain_allowed_usage = is_domain_allowed_usage and repeater_attempts_rate_limiter.allow_usage(domain)
+def rate_limit_repeater(domain, repeater_id):
+    limit_attempts = RATE_LIMIT_REPEATER_ATTEMPTS.enabled(domain, namespace=NAMESPACE_DOMAIN)
+    is_under_attempt_limit = repeater_attempts_rate_limiter.allow_usage(repeater_id) if limit_attempts else True
 
-    if global_repeater_rate_limiter.allow_usage() or is_domain_allowed_usage:
+    if global_repeater_rate_limiter.allow_usage() and is_under_attempt_limit:
+        allow_usage = True
+    elif repeater_rate_limiter.allow_usage(domain):
```

Comment:
For my understanding, `repeater_rate_limiter.allow_usage(domain)` (which was `is_domain_allowed_usage`) will prevent a domain from being rate limited even when the attempt limit is exceeded (when `limit_attempts` is true and `is_under_attempt_limit` is false). Is that the intended outcome?

Reply:
Yes that is intentional. We should only rate limit based on attempts (when `is_under_attempt_limit` is false) if the repeater (or domain in this case) is above the total wait times allowed, effectively only targeting slow repeaters.